### DISCUSSION TITLE
Is there a way to show less thumbnails when less than 5 slides?

### DISCUSSION THREAD

**Discussion by ARChilton** [igordanchenko/yet-another-react-lightbox#21] [Q&A] [answered]
I have only 3 images for one of my pages so the thumbnails repeat, is there a way to only show the number of thumbnails = to the number of slides if less than 5 slides?

The library is great btw, thanks!

**Comment by igordanchenko** [comment, marked as answer]
The Thumbnails plugin uses `carousel.preload` prop when determining how many thumbnails to show. So, in this case, you could set `carousel.preload` to 1. 

```jsx
<Lightbox
  open
  slides={slides}
  plugins={[Thumbnails]}
  carousel={{ preload: Math.min(Math.floor(slides.length / 2), 2) }}
/>
```

https://codesandbox.io/s/yet-another-react-lightbox-21-rjo2qd?file=/src/App.js

However, you'd still see repeating thumbnails when you have an even number of photos.

**Reply by davidburson** [reply]
Is there a way to avoid having repeating thumbnails?  If I have 4 slides, I need to see 4 thumbnails, not 3 (which means one thumbnail isn't shown) or 5 (which means a repeated thumbnail).

**Reply by igordanchenko** [reply]
The only way to avoid the repeating thumbnails in this case is to disable infinite carousel as follows: 

```jsx
slides={slides.slice(0, 4)}
carousel={{ finite: true, preload: 3 }}
```
